src/HeroComponent.js

```diff
@@ -0,0 +1,165 @@
+        if (key === localConfig.options[x].urlParam) {
+          const tempConfig = localConfig.options[x];
+
+          // To be cleaned up with a standardized model when all option types are defined.
+          if (tempConfig.type === 'radiogroup' || tempConfig.type === 'textfield') {
+            tempConfig.value = urlParams[key];
+          }
+        }
+      }
+    });
+
+    return localConfig
+  }
+
+  render() {
+    const urlParams = queryString.parse(this.props.location.search);
+    this.localConfig = this.copyUrlParamsToLocalConfig(this.localConfig, urlParams);
+    if (this.localConfig.afterUpdate) this.localConfig.afterUpdate();
+
+    return (
+        <React.Fragment>
+          <div className='heroComponent'>
+            <HeroTabs urlParams={urlParams} config={this.localConfig} {...this.props}>
+              {React.cloneElement(this.props.children, {...this.props.children.props, ...{config: this.localConfig}})}
+            </HeroTabs>
+          </div>
+        </React.Fragment>
+  )
+  }
+}
+
+class HeroTabs extends Component {
+  state = {activeIndex: 0, codeString: ''};
+
+  render() {
+    const tabContents = [
+      this.props.children,
+      <WebTab>{this.props.children}</WebTab>,
+      <ReactTab>{this.props.children}</ReactTab>,
+    ];
+    return (
+      <React.Fragment>
+        <TabBar
+            activeIndex={this.state.activeIndex}
+            handleActiveIndexUpdate={(activeIndex) => this.setState({activeIndex})}
+            className='catalog-hero-tab-bar'>
+          <Tab className={'hero-tab'}>
+            <span className='mdc-tab__text-label'>Demo</span>
+          </Tab>
+          <Tab className={'hero-tab'}>
+            <span className='mdc-tab__text-label'>Web</span>
+          </Tab>
+          <Tab className={'hero-tab'}>
+            <span className='mdc-tab__text-label'>React</span>
+          </Tab>
+        </TabBar>
+        <div className={'tab-container'}>
+          {tabContents.map((content, index) => {
+            if (this.state.activeIndex !== index) return null;
+            return (
+              <div className='tab-content' key={index}>
+                {content}
+              </div>
+            );
+          })}
+        <HeroOptionsComponent {...this.props} />
+        </div>
+
+      </React.Fragment>
+    )
+  }
+}
+
+const classesToRemove = [
+    ' mdc-ripple-upgraded--unbounded',
+    ' mdc-notched-outline--upgraded',
+];
+
+class WebTab extends Component {
+  state = {codeString: ''};
+
+  initRef = (ref) => {
+    let codeString = '';
+    if (ref) {
+      codeString = html.prettyPrint(ref.innerHTML);
+      classesToRemove.forEach((str) => codeString = codeString.replace(new RegExp(str, 'g'), ''));
+    }
+
+    this.setState({codeString});
+  };
+
+  render() {
+    return (
+        <React.Fragment>
+          <div style={{display: 'none'}} ref={this.initRef}>{this.props.children}</div>
+          <SyntaxHighlighter
```

Comment:
Would it make sense to pull out common properties between web and react and put it in a constant some where?